### PR TITLE
FAI-11836: Resets fail w/ HTTP 504

### PR DESCRIPTION
**Pull request by ted-faros**
## Description

Break non-incremental model reset queries into paginated delete by id queries

## Type of change
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/common/graphql-client.ts [hunk 1]
```diff
@@ -377,23 +385,49 @@ export class GraphQLClient {
...
-        [`delete_${model}`]: {
+      const query = {
+        [model]: {
           __args: {
             where: {
               _and: {...deleteConditions},
             },
           },
-          affected_rows: true,
+          id: true,
```
Comment (cjwooo) [inline destinations/airbyte-faros-destination/src/common/graphql-client.ts:395]: How does this line get translated into "return the id field"? (confused about the `true` portion)

**Review by cjwooo — APPROVED** [review]

**Review by ypc-faros — APPROVED** [review]
Can we also add this setting to the [spec](https://github.com/faros-ai/airbyte-connectors/blob/9eb8ad6b43ced6442a12d186703698304d18b8a0/destinations/airbyte-faros-destination/resources/cloud.json#L41) and wire it through?

File: destinations/airbyte-faros-destination/src/common/graphql-client.ts (hunk 1 above)
Comment (ted-faros) [reply destinations/airbyte-faros-destination/src/common/graphql-client.ts:395]: that is how the library works: https://www.npmjs.com/package/json-to-graphql-query#query-with-nested-objects

**Comment by ted-faros** [comment]
> Can we also add this setting to the [spec](https://github.com/faros-ai/airbyte-connectors/blob/9eb8ad6b43ced6442a12d186703698304d18b8a0/destinations/airbyte-faros-destination/resources/cloud.json#L41) and wire it through?

Added.  Can you take another look @ypc-faros ?

File: destinations/airbyte-faros-destination/resources/spec.json [hunk 2]
```diff
@@ -175,6 +175,12 @@
         "title": "Fail on source error",
         "description": "Fail the sync if the source has an error, even if it's not a fatal error.",
         "default": false
+      },
+      "reset_page_size": {
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/resources/spec.json:179]: nit: Add `"order": 17,`

**Review by ypc-faros — APPROVED** [review]

**Pull request merged**